### PR TITLE
[OpenCL] Fixes device name comparison stage_op_test 

### PR DESCRIPTION
**Pull request by jwlawson**
Changes the fixed '/device:GPU:0' expected device name string to the name returned by 'test/gpu_device_name()', as the device name could be '/device:SYCL:0' or '/device:GPU:0'.

Resubmission of #10682

### CODE REVIEW

**Review by yifeif — APPROVED** [review]
Approve as the same change has been approved in https://github.com/tensorflow/tensorflow/pull/10682. cc: @benoitsteiner

**Comment by yifeif** [comment]
@tensorflow-jenkins test this please

File: tensorflow/python/kernel_tests/stage_op_test.py [hunk 1]
```diff
@@ -102,8 +102,7 @@ def testColocation(self):
       with ops.device(gpu_dev):
         stager = data_flow_ops.StagingArea([dtypes.float32])
         y = stager.put([v])
-        self.assertEqual(y.device, '/device:GPU:0' if gpu_dev
-                                                   else gpu_dev)
+        self.assertEqual(y.device, gpu_dev)
```
Comment (benoitsteiner) [inline tensorflow/python/kernel_tests/stage_op_test.py]: This cause the following test failure on gpu:
======================================================================
FAIL: testColocation (__main__.StageTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/var/lib/jenkins/workspace/tensorflow-pull-requests-gpu/bazel-ci_build-cache/.cache/bazel/_bazel_jenkins/eab0d61a99b6696edb3d2aff87b585e8/execroot/workspace/bazel-out/local_linux-opt/bin/tensorflow/python/kernel_tests/stage_op_test.runfiles/org_tensorflow/tensorflow/python/kernel_tests/stage_op_test.py", line 105, in testColocation
    self.assertEqual(y.device, gpu_dev)
AssertionError: u'/device:GPU:0' != u'/gpu:0'

**Review by benoitsteiner — CHANGES_REQUESTED** [review]

**Comment by drpngx** [comment]
@jwlawson any luck with the test failure?

**Review by benoitsteiner — APPROVED** [review]

**Comment by benoitsteiner** [comment]
Jenkins, test this please.

**Pull request merged**